twoliter/embedded/imghelper

```diff
@@ -0,0 +1,398 @@
+#!/usr/bin/env bash
+# shellcheck disable=SC2034,SC2154,SC2164,SC2312
+
+# Store output artifacts in a versioned directory.
+OUTPUT_DIR="${OUTPUT_DIR}/${VERSION_ID}-${BUILD_ID}"
+
+FILENAME_PREFIX="${IMAGE_NAME}-${VARIANT}-${ARCH}-${VERSION_ID}-${BUILD_ID}"
+SYMLINK_PREFIX="${IMAGE_NAME}-${VARIANT}-${ARCH}"
+VERSIONED_SYMLINK_PREFIX="${IMAGE_NAME}-${VARIANT}-${ARCH}-${VERSION_ID}"
+FRIENDLY_VERSIONED_SYMLINK_PREFIX="${IMAGE_NAME}-${VARIANT}-${ARCH}-v${VERSION_ID}"
+
+OS_IMAGE_NAME="${FILENAME_PREFIX}"
+OS_IMAGE_SYMLINK="${SYMLINK_PREFIX}"
+OS_IMAGE_VERSIONED_SYMLINK="${VERSIONED_SYMLINK_PREFIX}"
+OS_IMAGE_FRIENDLY_VERSIONED_SYMLINK="${FRIENDLY_VERSIONED_SYMLINK_PREFIX}"
+
+DATA_IMAGE_NAME="${FILENAME_PREFIX}-data"
+DATA_IMAGE_SYMLINK="${SYMLINK_PREFIX}-data"
+DATA_IMAGE_VERSIONED_SYMLINK="${VERSIONED_SYMLINK_PREFIX}-data"
+DATA_IMAGE_FRIENDLY_VERSIONED_SYMLINK="${FRIENDLY_VERSIONED_SYMLINK_PREFIX}-data"
+
+BOOT_IMAGE_NAME="${FILENAME_PREFIX}-boot.ext4.lz4"
+BOOT_IMAGE_SYMLINK="${SYMLINK_PREFIX}-boot.ext4.lz4"
+BOOT_IMAGE_VERSIONED_SYMLINK="${VERSIONED_SYMLINK_PREFIX}-boot.ext4.lz4"
+BOOT_IMAGE_FRIENDLY_VERSIONED_SYMLINK="${FRIENDLY_VERSIONED_SYMLINK_PREFIX}-boot.ext4.lz4"
+
+VERITY_IMAGE_NAME="${FILENAME_PREFIX}-root.verity.lz4"
+VERITY_IMAGE_SYMLINK="${SYMLINK_PREFIX}-root.verity.lz4"
+VERITY_IMAGE_VERSIONED_SYMLINK="${VERSIONED_SYMLINK_PREFIX}-root.verity.lz4"
+VERITY_IMAGE_FRIENDLY_VERSIONED_SYMLINK="${FRIENDLY_VERSIONED_SYMLINK_PREFIX}-root.verity.lz4"
+
+ROOT_IMAGE_NAME="${FILENAME_PREFIX}-root.ext4.lz4"
+ROOT_IMAGE_SYMLINK="${SYMLINK_PREFIX}-root.ext4.lz4"
+ROOT_IMAGE_VERSIONED_SYMLINK="${VERSIONED_SYMLINK_PREFIX}-root.ext4.lz4"
+ROOT_IMAGE_FRIENDLY_VERSIONED_SYMLINK="${FRIENDLY_VERSIONED_SYMLINK_PREFIX}-root.ext4.lz4"
+
+CROSS_FACTORYDIR="/usr/share/factory"
+CROSS_SYSCONFIGDIR="/etc"
+CROSS_TMPFILESDIR="/usr/lib/tmpfiles.d"
+
+CA_CERTS_BUNDLE="${HOME}/certs/ca-bundle.crt"
+CA_CERTS_DEFAULT_BUNDLE="/usr/share/bottlerocket/ca-certificates/ca-bundle.crt"
+CA_CERTS_DEFAULT_LICENSES="/usr/share/licenses/ca-certificates"
+CA_CERTS_INSTALLDIR="${CROSS_FACTORYDIR}${CROSS_SYSCONFIGDIR}/pki/tls/certs"
+
+SBKEYS="${HOME}/sbkeys"
+
+VERITY_VERSION=1
+VERITY_HASH_ALGORITHM=sha256
+VERITY_DATA_BLOCK_SIZE=4096
+VERITY_HASH_BLOCK_SIZE=4096
```

Comment:
The verity stuff should probably just be a `do_verity_for_image` function that takes parameters. Unlike the `ca-cert` ones, I like these as globals because they're effectively constants that can't be changed without a massive compatibility break.

Please also look to `partyplanner` for inspiration on comments on both function calls and important constants.